packages/amplication-server/src/core/resource/resourceBtm.service.ts

```diff
@@ -0,0 +1,261 @@
+      originalResourceId: resourceId,
+      data: recommendations,
+    };
+  }
+
+  generatePromptForBreakTheMonolith(resource: ResourcePartial): string {
+    const entityIdNameMap = resource.entities.reduce((acc, entity) => {
+      acc[entity.id] = entity.name;
+      return acc;
+    });
+
+    const prompt: BreakTheMonolithPromptInput = {
+      dataModels: resource.entities.map((entity) => {
+        return {
+          name: entity.name,
+          fields: entity.versions[0].fields.map((field) => {
+            return {
+              name: field.name,
+              dataType:
+                field.dataType == EnumDataType.Lookup
+                  ? entityIdNameMap[field.properties["relatedEntityId"]]
+                  : this.dataTypeMap[field.dataType],
+            };
+          }),
+        };
+      }),
+    };
+
+    return JSON.stringify(prompt);
+  }
+
+  async preparePromptResultToBtmRecommendation(
+    promptResult: string,
+    resourceId: string
+  ): Promise<BreakTheMonolithRecommendationsResult> {
+    const promptResultObj =
+      this.mapToBreakTheMonolithPromptOutput(promptResult);
+
+    const recommendedResourceEntities = promptResultObj.microservices
+      .map((resource) => resource.dataModels)
+      .flat();
+
+    const duplicatedEntities = this.duplicatedEntities(
+      recommendedResourceEntities
+    );
+    const usedDuplicatedEntities = new Set<string>();
+
+    const originalResource = await this.getResourceDataForBtm(resourceId);
+    const originalResourceEntitiesSet = new Set(
+      originalResource.entities.map((entity) => entity.name)
+    );
+
+    return {
+      microservices: promptResultObj.microservices
+        .sort((microservice) => -1 * microservice.dataModels.length)
+        .map((microservice) => ({
+          name: microservice.name,
+          functionality: microservice.functionality,
+          dataModels: microservice.dataModels
+            .filter((dataModelName) => {
+              const isDuplicatedAlreadyUsed =
+                usedDuplicatedEntities.has(dataModelName);
+              if (duplicatedEntities.has(dataModelName)) {
+                usedDuplicatedEntities.add(dataModelName);
+              }
+              return (
+                originalResourceEntitiesSet.has(dataModelName) &&
+                !isDuplicatedAlreadyUsed
+              );
+            })
+            .map((dataModelName) => {
+              const entityNameIdMap = originalResource.entities.reduce(
+                (map, entity) => {
+                  map[entity.name] = entity;
+                  return map;
+                },
+                {} as Record<string, EntityPartial>
+              );
+
+              return {
+                name: dataModelName,
+                originalEntityId: entityNameIdMap[dataModelName]?.id,
+              };
+            }),
+        })),
+    };
+  }
+
+  mapToBreakTheMonolithPromptOutput(
+    promptResult: string
+  ): BreakTheMonolithPromptOutput {
```

Comment:
This is not promptOutopt...

maybe just "BreakTheMonolithOutput"

Reply:
it sounds like a variable name and not a function that does something

Reply:
I was talking about the return type, but same about the function name "mapToBreakTheMonolithOutput" (the prompt is the text we send to GPT, it is not the results that returned from it)

Reply:
That's why there is "output" there...but I can change it